lib/sycamore/sycamore/transforms/table_structure/table_transformers.py

```diff
@@ -1072,9 +1072,19 @@ def structure_to_cells(table_structure, tokens, union_tokens):
                             new_row = BoundingBox(row_rect.x1, row_rect.y2, row_rect.x2, rows[row_idx + 1]["bbox"][1])
                             rows.insert(row_idx + 1, {"bbox": new_row.to_list()})
                             for cell in cells:
-                                for idx, row_num in enumerate(cell["row_nums"]):
-                                    if row_num > row_idx:
-                                        cell["row_nums"][idx] += 1
+                                cell_row_nums = cell["row_nums"]
+                                if (
+                                    row_idx in cell_row_nums and row_idx + 1 in cell_row_nums
+                                ):  # if the cell spans the 2 rows increase the span
+                                    for idx, row_num in enumerate(range(len(cell_row_nums), 0, -1)):
+                                        cell_row_nums[idx] += 1
+                                        if row_num <= (row_idx + 1):
+                                            cell_row_nums.insert(idx, row_idx + 1)
+                                            break
```

Comment:
wait why not just `cell_row_nums.append(max(cell_row_nums) + 1)`? Also I think this iterator is crazy:
0, 3
1, 2
2, 1
3, 0

Reply:
Changed, I forgot that all span row numbers must be continuous